pages/scan-results/ScanResults.tsx

```diff
@@ -100,21 +128,42 @@ class ResultCard extends Component {
   }
 }
 
-class ScanResults extends Component {
-  constructor(props) {
+type SortMode = 'alphabetic' | 'size'
+
+type ScanResultsProps = {
+  router: NextRouter
+}
+
+type ScanResultsState = {
+  packages: Package[]
+  sortMode?: SortMode
+}
+
+const parsePackages = (value: string | string[] | undefined) =>
+  value
+    ? (typeof value === 'string' ? [value] : value)
+        .flatMap(str => str.split(','))
+        .map(str => str.trim())
+    : []
+
+class ScanResults extends Component<ScanResultsProps, ScanResultsState> {
+  constructor(props: ScanResultsProps) {
     super(props)
 
     const { router } = this.props
     const sortMode = router.query.sortMode
-    const packageStrings = router.query.packages
-    const packages = packageStrings
-      .split(',')
-      .map(str => str.trim())
-      .map(str => ({
-        promiseState: 'pending',
-        packageString: str,
-        ...parsePackageString(str),
-      }))
+    const packages = parsePackages(router.query.packages).map(str => ({
+      promiseState: 'pending' as const,
+      packageString: str,
+      ...parsePackageString(str),
+    }))
+
+    invariant(
+      sortMode === undefined ||
+        sortMode === 'alphabetic' ||
+        sortMode === 'size',
+      'invalid sort mode'
+    )
```

Comment:
Might be a better idea to use a default sort mode than throw an error to our users here.

Reply:
Done